src/app/modules/shared_models/token_model.nim

```diff
@@ -111,3 +111,12 @@ QtObject:
         return true
 
     return false
+
+  proc hasGas*(self: Model, chainId: int, requiredGas: float): bool {.slot.} =
+    for item in self.items:
+      if(item.getSymbol() == "ETH"):
+        for balance in item.getBalances().items:
+          if (balance.chainId == chainId):
+            if(balance.balance >= requiredGas):
+              return true
+    return false
```

Comment:
nit: personal preference but i feel it is more readable with guard to avoid the many indentation
e.g:

```
for item in self.items:
    if(item.getSymbol() != "ETH"):
        continue

    for balance in item.getBalances().items:
        if (balance.chainId != chainId):
            continue

        if(balance.balance >= requiredGas):
            return true
    
return false
```